stream/BidirectionalStream.java

```diff
@@ -118,22 +118,40 @@ private void close(@Nullable StatusRuntimeException error) {
         }
     }
 
-    public List<StatusRuntimeException> drainErrors() {
-        List<StatusRuntimeException> errors = new ArrayList<>(resCollector.drainErrors());
-        errors.addAll(resPartCollector.drainErrors());
+    void singleDone(UUID requestID) {
+        resCollector.remove(requestID);
+    }
+
+    void iteratorDone(UUID requestID) {
+        resPartCollector.remove(requestID);
```

Comment:
separate methods for separate collectors

Reply:
Just a fairly general comment, not limited to this change:

How aligned are the implementations of tx exception propagation in the 3 clients?

I'm pretty sure client-python had just one `done` method here, and client-nodejs doesn't appear to have any.

Reply:
yeah python has only one collector for all single and iterator, so it doesn't need split methods. I'm going to go back to update client-nodejs so it's in one of these two styles next (but won't release it)